include/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics.hpp

```diff
@@ -51,11 +61,42 @@ class Dynamics
 
     virtual void print(std::ostream& anOutputStream, bool displayDecorator = true) const = 0;
 
-    /// @brief              Obtain dynamical equations function wrapper (pure virtual)
+    /// @brief              Update the state derivative (pure virtual)
     ///
+    /// @param              [in] x A state vector
+    /// @param              [out] dxdt A state derivative vector
+    /// @param              [in] anInstant An instant
+
+    virtual void update(const StateVector& x, StateVector& dxdt, const Instant& anInstant) = 0;
+
+    /// @brief              Get dynamical equations function wrapper (pure virtual)
+    ///
+    /// @param              [in] aDynamicsArray A array of shared pointers to dynamics
+    /// @param              [in] anInstant An instant
     /// @return             std::function<void(const std::vector<double>&, std::vector<double>&, const double)>
 
-    virtual DynamicalEquationWrapper getDynamicalEquations() = 0;
+    static DynamicalEquationWrapper GetDynamicalEquations(
+        const Array<Shared<Dynamics>>& aDynamicsArray, const Instant& anInstant
+    );
+
+    const Shared<const Frame> gcrfSPtr_ = Frame::GCRF();
+
+   private:
+    /// @brief              Dynamical Equations
+    ///
+    /// @param              [in] x A state vector
+    /// @param              [out] dxdt A state derivative vector
+    /// @param              [in] t A step duration from anInstant to the next
+    /// @param              [in] aDynamicsArray A array of shared pointers to dynamics
+    /// @param              [in] anInstant An instant
+
+    static void DynamicalEquations(  // TBI: Unclear naming
```

Comment:
```suggestion
    static void DynamicalEquations(  // TBI: Unclear naming
```

Sorry, will remove, want to spend a bit more time on this